internal/providers/msentraid/msentraid.go

```diff
@@ -175,9 +158,10 @@ func (p Provider) getGroups(token *oauth2.Token) ([]info.Group, error) {
 		if !ok || name == nil {
 			id := msGroup.GetId()
 			if id == nil {
+				unknown := "Unknown"
 				id = &unknown
 			}
-			slog.Warn(pp.Sprintf("Could not get displayName from group object (ID: %s) found: %v", *id, *msGroup))
+			slog.Warn(pp.Sprintf("Could not get displayName from group object (ID: %s) found: %v", *id, msGroup))
 			return nil, errors.New("could not parse group name")
 		}
```

Comment:
Could we use `msGroup.GetDisplayName()` here instead of `msGroup.GetBackingStore().Get("displayName")` to get rid of this check as well? Do we not trust the graph API to return the types it's supposed to?